### PR TITLE
When you Switch HW wallet, removeAccount not worked as you think

### LINKED ISSUE #9937: syncWithAddresses not remove accounts from cachedBalances
After you import some addresses from ledger, when your import addresses from trezor, the old accounts not removed from `cachedBalances`

### PR DESCRIPTION
**Pull request by codeoneline**
Fixes: #9937 

Explanation:  
file: `app/scripts/lib/account-tracker.js`
func: `syncWithAddresses (addresses)`
```
  syncWithAddresses (addresses) {
    ...
    this.addAccounts(accountsToAdd)             // error, it add new accouts in a async way,
    this.removeAccount(accountsToRemove)  // it remove accounts in a sync way, this should move before the async operate.
  }
```
`this.addAccounts(accountsToAdd)` =>

 `this.store.updateState({ accounts })` // but, will override after `this._updateAccounts()`
 `this._updateAccounts()`  => 

 `const { accounts } = this.store.getState()`
`await this._updateAccountsViaBalanceChecker` =>

`const { accounts } = this.store.getState()`  // oldAccounts + newAccounts
`ethContract.balances` => // in callback, will update oldAccounts + newAccounts balance

      `this.store.updateState({ accounts })`  // this will override the this.removeAccount in a async way

Then `getSelectedAccount() `will be undefine, after you `removeAccounts`, so in `eth-overview.js` replace 
`const selectedAccount = useSelector(getSelectedAccount)` with
  ```
const address = useSelector(getSelectedAddress)
  let selectedAccount = useSelector(getSelectedAccount)
  if (!selectedAccount) {
    selectedAccount = {address: address, balance: '0x0'}
  }
```
to avoid crash

I have read the CLA Document and I hereby sign the CLA

### CODE REVIEW

**Comment by codeoneline** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by brad-decker** [comment]
@codeoneline could you fix up the lint issues while you're waiting for review? Thanks again for your contributions!

**Review by Gudahtt — CHANGES_REQUESTED** [review]
Thanks for the contribution! Most of these changes make sense to me, but I did have questions about the `eth-overview.js` part.

>Then getSelectedAccount() will be undefine, after you removeAccounts

Since `this.preferencesController.syncAddresses(addresses)` is called before `this.accountTracker.syncWithAddresses(addresses)`, it shouldn't be possible for `getSelectedAccount()` to be `undefined` anymore. The `syncAddresses` call should updated `selectedAddress` such that it's not among the accounts being removed.

Does that make sense? I think if we discard the `eth-overview.js` changes and rebase the `account-tracker.js` changes on `develop`, this PR will be good to merge. Let me know what you think!

**Review by codeoneline — COMMENTED** [review]
> Thanks for the contribution! Most of these changes make sense to me, but I did have questions about the `eth-overview.js` part.
> 
> > Then getSelectedAccount() will be undefine, after you removeAccounts
> 
> Since `this.preferencesController.syncAddresses(addresses)` is called before `this.accountTracker.syncWithAddresses(addresses)`, it shouldn't be possible for `getSelectedAccount()` to be `undefined` anymore. The `syncAddresses` call should updated `selectedAddress` such that it's not among the accounts being removed.
> 
> Does that make sense? I think if we discard the `eth-overview.js` changes and rebase the `account-tracker.js` changes on `develop`, this PR will be good to merge. Let me know what you think!

Othewise, It will give an error message like this (I've tried on v8.0.10 and v9.1.0):
`
TypeError: Cannot read property 'balance' of undefined
  at isBalanceCached (chrome-extension://opkjncnmjekcjbhhggfiaccpeokadple/ui.js:286327:85)
  at useSelectorWithStoreAndSubscription (chrome-extension://opkjncnmjekcjbhhggfiaccpeokadple/ui.js:197075:23)
  at useSelector (chrome-extension://opkjncnmjekcjbhhggfiaccpeokadple/ui.js:197151:12)
  at EthOverview (chrome-extension://opkjncnmjekcjbhhggfiaccpeokadple/ui.js:243241:55)
  at renderWithHooks (chrome-extension://opkjncnmjekcjbhhggfiaccpeokadple/ui.js:180425:18)
  at mountIndeterminateComponent (chrome-extension://opkjncnmjekcjbhhggfiaccpeokadple/ui.js:182959:13)
  at beginWork$1 (chrome-extension://opkjncnmjekcjbhhggfiaccpeokadple/ui.js:184327:16)
  at HTMLUnknownElement.callCallback (chrome-extension://opkjncnmjekcjbhhggfiaccpeokadple/ui.js:164501:14)
  at Object.invokeGuardedCallbackDev (chrome-extension://opkjncnmjekcjbhhggfiaccpeokadple/ui.js:164550:16)
  at invokeGuardedCallback (chrome-extension://opkjncnmjekcjbhhggfiaccpeokadple/ui.js:164605:31)
`


If call `this.removeAccount` first, It may remove the current selected account from cache, but `getSelectedAccount()` still want to get the old account's balance from cache, then it crashed.

When first import an account from ledger, current account is a ledger address,  then switch to trezor, `syncWithAddresses(addresses)`, the selected ledger address is not in the addresses of a trezor. so it will remove all ledger's address, but `getSelectedAccount()` still want to get the ledger address's balance, then it crashed.

When I add  `this.preferencesController.syncAddresses(accountsToTrack);`
```
async connectHardware(deviceName, page, hdPath) {
...
    this.preferencesController.syncAddresses(accountsToTrack);
    this.accountTracker.syncWithAddresses(accountsToTrack);
    return accounts;
}
```
in file `metamask-controller.js`, it works fine!! 
Please check the new commit!!

File: app/scripts/metamask-controller.js [hunk 1]
```diff
@@ -1271,6 +1271,7 @@ export default class MetamaskController extends EventEmitter {
         oldAccounts.concat(accounts.map((a) => a.address.toLowerCase())),
       ),
     ];
+    this.preferencesController.syncAddresses(accountsToTrack);
```
Comment (Gudahtt) [inline app/scripts/metamask-controller.js:1274]: I don't think we want this :thinking: The accounts here are being previewed during the hardware connect process. This includes the current "page" that you're looking at. We don't want to add identities for each of these.

**Comment by brad-decker** [comment]
@codeoneline did you want to carry this through to completion? You've had a few rounds of feedback and just want to make sure you saw @Gudahtt 's message.

**Comment by brad-decker** [comment]
Thank you for your contribution @codeoneline. In an attempt to keep our PR queue clean and focus our review time we are closing pull requests that become stale. This doesn't mean we don't want your contribution, and we encourage you to reopen the PR once the changes/feedback requested is completed. Thank you again!

**Pull request closed without merging**